pkg/cli/cli.go

```diff
@@ -35,6 +35,8 @@ type CommandOptions struct {
 	IOStreams
 	Args   []string
 	Config Config
+	// Retrieves the value of the environment variable named by the key.
+	LookupEnv func(key string) (string, bool)
```

Comment:
I get the feeling this variable only exists to allow testing. If that's the case you could take advantage of the fact that Go tests are never run in parallel unless you mark them so, which means it would be safe to set the environment variable before the test and resetting it back to its original state after.
Environment variables are  only available in the process that sets them and in any forked processes, which is why you need the guarantee that the test won't run concurrently with anything else and that no forks will be made.
This is more a suggestion than a request, if you take it make sure to add a comment in the test explaining the assumption about non-concurrency and warning against enabling it for that particular test.

Reply:
good point, used t.Setenv https://pkg.go.dev/testing#B.Setenv instead, documentation mentions that tests using this method cannot be executed in parallel.